### PR TITLE
RFC: Long term future of applications

### PR DESCRIPTION
**Pull request by tijmenb**
👉[Rendered version](https://github.com/alphagov/govuk-rfcs/blob/long-term-future-of-apps/rfc-095-long-term-future-apps.md)

### CODE REVIEW

**Comment by rubenarakelyan** [comment]
We need to move specialist-publisher and travel-advice-publisher to go through the usual email-alert-service route, and then we can retire the service altogether and merge it into email-alert-api. This was the original plan but we didn't have time to do it.

File: rfc-095-long-term-future-apps.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+## Proposal
+
+| Name | Description | Rough action plan |
+| -- | -- | -- |
+| calculators | Serves the Child benefit tax calculator on GOV.UK | Merge into smart-answers |
+| calendars | Serves /bank-holidays and /when-do-the-clocks-change on GOV.UK | Merge into custom frontend |
+| ckanext-datagovuk | Extension for use with datagovuk_publish | Retire in favour of datagovuk_find/datagovuk_publish? |
+| collections-publisher | Publishes step by steps, /browse pages, and legacy /topic pages on GOV.UK | While topics and browse pages will be removed, this can still be the app for curating collections. We might move Whitehall's document collections into this app |
+| content-audit-tool | Deprecated application to audit content | Retire, as it's not used |
+| content-performance-manager | Data warehouse that stores content and content metrics to help content owners measure and improve content on GOV.UK | Rename to "data warehouse" |
```
Comment (erino) [inline rfc-095-long-term-future-apps.md]: I thought this was going to be called `content-data`?

Comment (tijmenb) [reply rfc-095-long-term-future-apps.md]: Fixed!

File: rfc-095-long-term-future-apps.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+| service-manual-frontend | Serves the Service Manual and Service Toolkit on GOV.UK | Parts of the frontend could be merged into government-frontend, and other apps |
+| service-manual-publisher | Publishes the Service Manual on GOV.UK | Custom application that could long term be replaced by content-publisher  |
+| specialist-publisher | Publishes specialist documents on GOV.UK | Retire in favour of content publisher |
+| static | GOV.UK static files and resources | Retire in favour of [the components gem][cg] |
+| travel-advice-publisher | Publishes foreign travel advice on GOV.UK | Retire in favour of content publisher |
+| whitehall | Publishes government content on GOV.UK | Retire in favour of content publisher, organisations publisher, and collections publisher, custom frontend (history pages) |
+
+[cg]: https://github.com/alphagov/govuk_publishing_components
+
+### No changes expected
```
Comment (thomasleese) [inline rfc-095-long-term-future-apps.md]: There are a few apps in this section which could be renamed in a similar way to the renames listed in the section above:

- collections to collections-frontend
- feedback to feedback-frontend
- support to support-admin
- smart-answers to smart-answers-frontend

(to name a few)

Should these be added to the section above?

File: rfc-095-long-term-future-apps.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+| local-links-manager | Manages local links from local authorities on GOV.UK | No change expected |
+| publishing-api | API to publish content on GOV.UK | No change expected |
+| release | Helps deploying to GOV.UK | No change expected |
+| router | Router in front on GOV.UK to proxy to backend servers on the single domain | No change expected |
+| search-admin | Admin for GOV.UK search | No change expected |
+| short-url-manager | Tool to request, approve and create short URL redirects on GOV.UK | No change expected |
+| signon | Single sign-on service for GOV.UK | No change expected |
+| support-api | API for processing GOV.UK named requests and anonymous feedback | No change expected |
+| info-frontend | Serves /info pages to display user needs and performance data about a page on GOV.UK | No change expected |
+| mapit | GOV.UK fork of Mapit, a web service to map postcodes to administrative boundaries | No change expected |
```
Comment (tijmenb) [inline rfc-095-long-term-future-apps.md]: @sihugh could we long-term use a hosted version of Mapit?

Comment (rubenarakelyan) [reply rfc-095-long-term-future-apps.md]: There is this: https://mapit.mysociety.org/pricing/. We're just talking about having to do quarterly updates of MapIt data and how much of a pain it is. Leaving this to a service to do for us could save a lot of time and grief.

**Comment by martinlugton** [comment]
I've been thinking about ckan.

On ckanext-datagovuk it may make more sense in the long-run to stay closer to core ckan for the publish backend, rather than to progress with the partially-completed datagovuk_publish.

Once we've upgraded to ckan to 2.7 we'll be using a mostly vanilla install with a clear upgrade path. Whereas if we progress with datagovuk_publish we have to put in a lot of work to build harvesting ourselves, and to implement any new features that the data community expects.

The main challenges of this 'stay close to core ckan for publishing' approach are:
- We'd need to be comfortable working with python, and a slightly unusual python architecture.
- I don't know how easy it would be in practice for us to contribute back to core ckan.

Maybe we can do the 2.7 upgrade, monitor it for a bit, and then make a decision.

File: rfc-095-long-term-future-apps.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+| calendars | Serves /bank-holidays and /when-do-the-clocks-change on GOV.UK | Merge into custom frontend |
+| ckanext-datagovuk | Extension for use with datagovuk_publish | Retire in favour of datagovuk_find/datagovuk_publish? |
+| collections-publisher | Publishes step by steps, /browse pages, and legacy /topic pages on GOV.UK | While topics and browse pages will be removed, this can still be the app for curating collections. We might move Whitehall's document collections into this app |
+| content-audit-tool | Deprecated application to audit content | Retire, as it's not used |
+| content-publisher | WIP - Future publisher of content on GOV.UK | Expand to publish most content |
+| datagovuk_find | Beta version of Find Data | Merge into finder-frontend? |
+| datagovuk_publish | Beta version of publish data | Retire in favour of content publisher? |
+| email-alert-service | Message queue consumer that triggers email alerts for GOV.UK | Merge into email-alert-api, as it's only task is to take things from the message queue and forward it there |
+| licence-finder | Serves licence pages on GOV.UK | Retire |
```
Comment (timblair) [inline rfc-095-long-term-future-apps.md]: It's unclear what the future is for licence-finder. We have some responsibility under the EU Service Directive to provide a central way for users to discover where they _can_ apply for licences. I'd say it's likely that licence-finder as-is will be retired, but additional functionality might be required in (e.g.) local links manager to support what we do need to do.

File: rfc-095-long-term-future-apps.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+### Apps that need a Rename
+
+| Name | Description | Rough action plan |
+| -- | -- | -- |
+| bouncer | Handles traffic for sites that have transitioned to GOV.UK | Rename to "transition-redirector" |
+| collections | Serves the new navigation pages, browse, topic and services and information pages on GOV.UK | Rename to "collections-frontend" |
+| content-performance-manager | Data warehouse that stores content and content metrics to help content owners measure and improve content on GOV.UK | Rename to "content-data" |
+| feedback | Serves contact pages on GOV.UK | Rename to "feedback-frontend" |
+| finder-frontend | Serves search pages for GOV.UK | Rename to "search-frontend". |
+| frontend | Serves the homepage, transactions and some index pages on GOV.UK | Rename to "custom frontend" |
```
Comment (timblair) [inline rfc-095-long-term-future-apps.md]: <nitpick> custom frontend --> custom-frontend </nitpick>

File: rfc-095-long-term-future-apps.md (hunk 3 above)
Comment (timblair) [reply rfc-095-long-term-future-apps.md]: I'll all for using a hosted version. I'm pretty sure we could save £££ even with going for the £300/month option, when you take in to account both existing hosting costs, plus time taken to maintain / update the dataset.

Comment (sihugh) [reply rfc-095-long-term-future-apps.md]: Last I looked, there were a couple of small problems:
 - hosted mapIt didn't have official names for councils (might be fixed now)
 - we've added local authority slugs to our mapIt to help with the postcode -> council page journey which would need some unravelling

I'd love a hosted version for government.  I've had at least 5 questions from random service teams about what we use.

Comment (martinlugton) [reply rfc-095-long-term-future-apps.md]: I'm late to the party on this, with old information, but the latest thinking on mapit is in this doc

https://docs.google.com/document/d/1VyxP5qSZzx5cFL4CQk-k0CKU3Z342VP_fky3D_8pebo/edit?usp=sharing_eil&ts=5c07e833

A hosted service was attractive, but there was a concern about the SLA that Mysociety could offer.

**Pull request merged**